packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -332,6 +332,9 @@ export class Summarizer extends EventEmitter implements ISummarizer {
                 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                 const startP = this.start(this.runtime.clientId!, runCoordinator);
                 startP.then(async (runningSummarizer) => {
+                    if (runningSummarizer.refreshSummaryAckLock !== undefined) {
```

Comment:
This looks wrong to me. Why not await for it inside of summarizeOnDemand?
I'd not make public refreshSummaryAckLock - it should stay private.

Reply:
It was due to the fact that summarizeOnDemand  is not async and I'd need to wait for the lock.  Do you have an alternative ?